OOP-52 Implement pagination in materials handler and materials frontend management

StanleighMorales opened the pull request:
https://oopfinalscs22agroup3.atlassian.net/browse/OOP-52

Implement pagination in materials handler and materials frontend management:
added SSR pagination on materials
search and filter functions still works
updated the color coding of the status

StanleighMorales commented:
/genimi-review

StanleighMorales commented:
/gemini-review

Neil-urk12 commented on Backend/cmd/web/main.go:
```diff
@@ -15,6 +15,7 @@ import (
 
+
 	_ "oop/docs" // load API docs generated by Swag CLI
 
```
check this lang if it bugs or no
If it bugs then resolve if not then no

Neil-urk12 commented on Backend/internal/handlers/materials_handlers.go:
```diff
@@ -33,11 +33,12 @@ func (h *MaterialHandlers) RegisterMaterialRoutes(r fiber.Router) {
-	materialsGroup.Get("/:id", h.GetMaterialHandler)       // GET /api/materials/{id}
-	materialsGroup.Post("/", h.CreateMaterialHandler)      // POST /api/materials
-	materialsGroup.Put("/:id", h.UpdateMaterialHandler)    // PUT /api/materials/{id}
-	materialsGroup.Delete("/:id", h.DeleteMaterialHandler) // DELETE /api/materials/{id}
```
Check this

Neil-urk12 commented on Backend/internal/handlers/materials_handlers.go:
```diff
@@ -33,11 +33,12 @@
+	materialsGroup.Get("/paginated", h.GetPaginatedMaterialsHandler) // GET /api/materials/paginated?page=1&limit=10
+	materialsGroup.Get("/:id", h.GetMaterialHandler)                 // GET /api/materials/{id}
+	materialsGroup.Post("/", h.CreateMaterialHandler)                // POST /api/materials
+	materialsGroup.Put("/:id", h.UpdateMaterialHandler)              // PUT /api/materials/{id}
```
Properly handle the errors

Neil-urk12 commented on Backend/internal/repositories/materials_repository.go:
```diff
@@ -15,6 +15,7 @@ type MaterialRepository interface {
+	GetPaginated(page, limit int, searchTerm, category, supplier, status string) ([]models.Material, int64, error)
 }
 
 // materialRepository implements the MaterialRepository interface
```
Check if this might lead to errors or no

Neil-urk12 commented on Frontend/src/stores/materials.ts:
```diff
@@ -14,8 +14,7 @@
-import { useSearch } from 'src/utils/useSearch';
-import { operationNotifications } from 'src/utils/notifications';
+import type { QTableProps } from 'quasar'
 
```
do this

Neil-urk12 commented on Frontend/src/stores/materials.ts:
```diff
@@ -50,74 +49,48 @@
-    } finally {
-      isLoading.value = false
-    }
+    await onRequest({ pagination: pagination.value })
```
do this
const rowsPerPage = ref(10)

Neil-urk12 commented on Frontend/src/stores/materials.ts:
```diff
@@ -141,20 +114,17 @@
    * @returns {MaterialRow[]} The filtered array of material rows.
    */
   const filteredMaterialRows = computed(() => {
-    const filtered = materialRows.value.filter(row => {
```
optional but it's better if you do this

StanleighMorales commented:
improve error handler
added some reactive states variables 
better performance when searching
Updated the pagination ref initialization to use this constant instead of the magic number 10
Updated the default value in the destructuring assignment within the onRequest function to use the constantUpdated the pagination ref initialization to use this constant instead of the magic number 10
Updated the default value in the destructuring assignment within the onRequest function to use the constant
